core/trino-parser/src/main/java/io/trino/sql/tree/StringLiteral.java

```diff
@@ -13,19 +13,20 @@
  */
 package io.trino.sql.tree;
 
-import io.airlift.slice.Slice;
-
-import java.util.Objects;
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
 import java.util.Optional;
 
-import static io.airlift.slice.Slices.utf8Slice;
+import static java.nio.charset.StandardCharsets.UTF_8;
 import static java.util.Objects.requireNonNull;
 
 public class StringLiteral
         extends Literal
 {
-    private final String value;
-    private final Slice slice;
+    private static final Charset CHARSET = UTF_8;
```

Comment:
inline

(defining  it here would make sense, if it could be something else in some other class, but the fact that values are utf-8 encoded is a global property of Trino)